index.js

```diff
@@ -142,8 +143,10 @@ function parse (args, opts) {
     var next
     var value
 
+    if (configuration['collect-unknown-options'] && !arg.match(negative) && !hasAllShortFlags(arg) && !hasFlagsMatching(arg, /^-+([^=]+?)=[\s\S]*$/, /^-+no-([^=]+?)=[\s\S]*$/, /^-+([^=]+?)$/, /^-+no-([^=]+?)$/, /^-+([^=]+?)-$/, /^-+([^=]+?)\d+$/, /^-+([^=]+?)\W+.*$/)) {
```

Comment:
let's pull the bits after the `&&` into a helper called `isUnknownOption`; these regexes are making my eyes bleed, and will read better if they're assigned to variables in a helper function that describe their purpose.

Reply:
@bcoe I've implemented this requested change.  I also added some comments in the `isUnknownOption` and `hasAllShortFlags` explaining the logic a bit.